### ISSUE TITLE
Ignite damage is able to stack infinitely

### ISSUE THREAD

**Issue by jlldarocha** [gamefreedomgit/Maelstrom#3487]
**Description:**
Ignite isnt "spending" the previous ignite damage on its usual tick duration, every critical hit that lands before the first tick only adds to the pool, and the damage never happens, until you stop critting. 
This leads to extreme ignite damage if you land several crits in a row, and obscene combustions, way higher than what was possible with T13hc gear on retail, on 350s gear.

**How to reproduce:**
Land several critical strikes in a row, never going more than 2s(ignite tick frequency) without critting again on the target. Watch as ignite doesnt do damage until you stop critting inside that interval and only keeps building.

**Ignite goes 6 seconds on target without doing damage, only adding to the ignite pool, and then doing a 43k ignite:**

![ignite not doing damage](https://github.com/gamefreedomgit/Maelstrom/assets/127895081/cebfce41-827e-42ce-b291-35acfd9a98a7)

### 
**Extreme test with 100% crit gear on test server:**

**37 crits in a row, aaplying ignite 37 times, and ticking 3, with an average damage of 851k on ignite.**

![37 crits in a row](https://github.com/gamefreedomgit/Maelstrom/assets/127895081/e0358609-2d85-4634-978f-a544c4d70ef9)

**Another extreme test, this time over 300 crits in a row just to show ignite doesnt tick over 5 minutes, until i stop landing critical strikes before the 2s mark.**

![300](https://github.com/gamefreedomgit/Maelstrom/assets/127895081/fe36db7e-f72d-4477-9565-17da2ce75217)

**How it should work:**
Ignite Should only refresh and add to the pool if the duration is under 4 seconds. 

EJ 4.3 mage post, ignite mechanics
![image](https://github.com/gamefreedomgit/Maelstrom/assets/127895081/4fb4a3a8-5de7-4547-ab5f-f1dc8855598b)

More info on [Fire mage EJ guide/discussion:](http://web.archive.org/web/20120207223106/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/)

**Comment by SexyPancake** [comment]
Ignite being refreshed before proccing the 1st DoT of the extended Ignite is the only part of that github that is currently a bug.
Because of that you are indeed able to stack abnormal Combustion but on the flip side also losing a massive amount of Ignite as testing with 100% crit is out of the actual representation of ingame. The rest of the information is misleading (not attributing intention but it is still). 

This is the full information on Ignite Munching that represent the normal behavior : 
_______________________________________________________________________________________________________________________________________
**Crit: Ignite Munching**

http://web.archive.org/web/20120207223106/http://elitistjerks.com/f75/t110326-cataclysm_fire_mage_compendium/

Thanks to [mysteltainn](http://web.archive.org/web/20120207223106/http://elitistjerks.com/members/38488-mysteltainn/) for writing this section below.

The oft-referred to "Ignite Munching" is a phenomenon that results in less (or in rare cases, more) Ignite damage than would be otherwise predicted.

There are three reasons for its occurance: simultaneous criticals, criticals that occur shortly before an ignite tick, and criticals that occur when the ignite debuff's remaining duration is greater than four seconds. **The two former causes are caused by old, nigh-unfixable limitations within the game engine, and the latter is to do with a recent change in how DoT effects are refreshed.**

As the name states, simultaneous criticals is when two spells crit simultaneously on a given target - usually, these times are logged as the same to a thousanth of a second. Most commonly, the spells responsible are either a fireball with a queued hot streak, or a scorch with a queued fire blast. The result is that only the second logged spell's damage being considered for ignite - the first spell's damage is completely ignored with respect to ignite. In a log, this might look like:


[00:16.414] Spell Crit #1 on Target
[00:16.414] Spell Crit #2 on Target
Here, ignite would be calculated only considering Spell Crit #2's damage.

An obvious solution would be to prefer against queuing spells with identical travel times, but since a fire mage's myriad or DoTs can crit, it will still happen. Indeed, with relatively low crit rates (compared to late-WotLK where this munching was the majority of lost ignite damage) this is not as prominent as it was - but it is still a hefty chunk of damage lost.

As long as simultaneous events are problematic in the game as a whole, this type of munching will always occur. However, it could be alleviated by changing the travel time on certain spells - if pyroblast was slightly slower or faster than a fireball, for example.

The next cause of munching is based on ignite's 'talent delay' - a small delay between a given spell critical and the actual ignite application. In a log, ignite applying might look something like this:


[00:23.409] Spell Crit on Target
[00:24.158] Ignite Refreshed on Target
The time between the actual crit and the ignite applying is the server verifying that, yes, your character has the talent necessary before actually applying the debuff. Proc-based talents from all classes suffer this ~0.5s delay. This delay isn't a huge problem for other class-based procs, but strange things with ignite happen with certain sequences of events:


[00:13.872] Spell Crit on Target
[00:14.381] Ignite Tick
[00:14.461] Ignite refreshed on Target
**In this situation, the Ignite Tick is a 'free' tick - since it falls between a valid crit refreshing the ignite (verifying that the mage has the ignite talent and then calculating its damage pool) and the actual refresh event, the damage it deals isn't subtracted from the total - or, more accurately, the total is 'rolled back' to when the crit occured.**

There is a rarer variation of this where the following happens:


[00:30.064] Spell Crit #1 on Target
[00:30.097] Spell Crit #2 on Target
[00:30.512] Ignite refreshed on Target (from Spell Crit #1)
[00:30.902] Ignite Tick
[00:30.902] Ignite refreshed on Target (from Spell Crit #2)
The difference in this situation is that there's a crit and refresh sandwiching the crit in the last example. The result is that the Ignite Tick does damage as calculated considering Spell Crit #1 - then 'rolls back' the total to before either crit occured, and adds Spell Crit #2 instead.

Both of these scenarios are unavoidable, especially with DoTs ticking. I believe the first bug adds a non-trivial amount of damage, but I can't say much for the second - it is quite rare, and a favorable outcome when the alternative would be the next type of munching.

The last type of munching is a side-effect of the new Cataclysm DoT system.

Ignite extends past four seconds because of the new DoT refresh mechanic. This mechanic is present on other player-cast DoTs like living bomb and corruption, and was implemented to make clipping the DoT impossible. For example, if, before 4.0.1, if you were to refresh living bomb right before it was about to tick, perhaps at 3.1s, living bomb would refresh to 12s, and would tick 3s later at 9s - the time between those two ticks was effectively 5.9s; after 4.0.1, the duration of living bomb would refresh to 12.1s, tick at 12s, and continue to tick as normal.

Following this new system, ignite's behavior is as follows:
- Ignite's initial application will be 4s in duration: 2 ticks, one at 2s, one at 0s. Ignite's total damage pool is divided between these two ticks.
- Any refreshes at any time will refresh ignite to 6s in (max) duration (current duration is ignite's current duration mod 2 plus 4s): 3 ticks, one at 4s, one at 2s, one at 0s. Ignite's damage pool is divided evenly between these three ticks.
- Ignite must be below 4s in current duration to accept any new refresh events.

To Illustrate:

[00:54.436] Spell Crit #1 on Target
[00:54.819] Ignite Refreshed on Target
[00:55.163] Spell Crit #2 on Target
[00:56.354] Ignite Tick on Target
Spell Crit #2 is munched in this situation.

**This means that ignite will, on average, accept only one crit every two seconds. This is the other major source of ignite munching - DoTs critting at inopportune times and multiple crits in a small time-frame.**

It is possible to diminish this effect by attempting to space spells out, but DoTs will make this form of munching inevitable.

This 'limitation,' while seemingly arbitrary, is more about how ignite evenly splits its damage between its ticks and how the new system requires a DoT tick every period without interruption. To further refresh ignite would require that either the duration be set to 6s or the duration to arbitrarily lengthen as crits occured.

Two possible solutions might be to only add the new damage to ignite without affecting its duration when the current duration is over 4s, or to have its ticks not spread evenly, but each one representing only crits that happened in the past four seconds. As most other systematic problems in WoW, I'm keen to believe this is either old and hard-to-update, or not a high priority (or more likely, both).

There is one final anomaly. When spells simultaneously crit on a target without ignite present:


[00:16.414] Spell Crit #1 on Target
[00:16.414] Spell Crit #2 on Target
[00:16.938] Ignite #1 Applied to Target
[00:16.938] Ignite #2 Applied to Target
The resulting ignite, as expected, only considers Spell Crit #2. However, because of the simultaneous ignite application events, it ticks as hard as a 4s ignite would (damage pool/2) over 6s - an extra tick - 50% more damage than expected.

Do note that this is still a damage loss so long as Spell Crit #1 is more than 50% of Spell Crit #2.

**Comment by jlldarocha** [comment]
you dont lose any ignite, as soon as you stop critting it deals the stacked damage. the only thing its doing is making combustions way powerful than what it should. 
the first screenshot is with 350s gear, the 100% crit is just to illustrate it doesnt tick at all, as long as you keep landing critical strikes, i spend 5 minutes before the first ignite damage happening.

**Comment by SexyPancake** [comment]
If there was no ignite lost we wouldn't see any ignite munching unless I'm missing something obvious right now. The game can't register multiple crits in a small frame window. The fact that you can refresh the DoT before the first tick at 4sec doesn't mean that every crit will extend it. Really simple to test with combustionhelper out, without 100% crit you will see ignite munching and you even worst at 100% crit if you align a fireball crit to land with a pyroblast crit.

**Comment by jlldarocha** [comment]
every crit is extending it. On my testing, i crit for over 5 minutes straight, and no damage was lost, but it turned into a 5m ignite. i can hop on the test server with you if you wanna check. 

As long as you crit before the first ignite tick(between 6 and 4 seconds) it wont crit, and keeps adding until you dont refresh.

**Comment by UltraNix** [comment]
I just tested and the duration of Ignite is always < 4 sec for me (never refreshed with 6 sec). What am I missing?

**Comment by UltraNix** [comment]
@SexyPancake @Pornoodemon Please, poke me on Discord once you have time.

**Comment by bonfax4** [comment]
How accurate is that ignite info ? I've seen situations where threads failed to update certain mechanics later on. ( Paladin ghost strike in wotlk is a good example that comes to mind, if you follow the comments you realize blizzard fixed it without a proper bluepost or changelog, but the paladin post still had it up). 

> 
> To Illustrate:
> 
> [00:54.436] Spell Crit https://github.com/gamefreedomgit/Maelstrom/issues/1 on Target
> [00:54.819] Ignite Refreshed on Target
> [00:55.163] Spell Crit https://github.com/gamefreedomgit/Maelstrom/issues/2 on Target
> [00:56.354] Ignite Tick on Target
> Spell Crit https://github.com/gamefreedomgit/Maelstrom/issues/2 is munched in this situation.
> 
I can already see this part being wrong or outdated just watching[ this video](https://www.youtube.com/watch?v=_9jA3OGOlpM) and seeing @6 seconds when ignite has a bout 4.2-4.1sec left, a crit landed and still incurred a refresh. Solely relying on EJ main posts can sometimes be innacurate.

**Comment by bonfax4** [comment]
Dev came to the same conclusion over at twinstar that crits over 4sec still refresh to teh same duration but get the added damage.

https://cata-twinhead.twinstar.cz/?issue=7813 Translated

> They had a mistake on EJ + they were missing something for munching.
> 
> 1) If the mage critne again on a target where it has an extended Ignite (over 4 seconds), this new Ignite should be added to the pool on the target and the timer to the next tick should be unchanged (as with other DoT according to the DoT system). Currently, any crit above the 4-second duration of Ignit is ignored and not added to the pool.
> 
> [21:46:37.466] Logoz Pyroblast Alysrazor *45238*
> [21:46:37.990] Logoz Pyroblast Alysrazor *5355*
> [21:46:38.152] Alysrazor afflicted by Ignite from Logoz
> [21:46:38.823] Logoz Pyroblast Alysrazor *5355*
> [21:46:38.823] Logoz Pyroblast Alysrazor *46823*
> [21:46:39.412] Alysrazor's Ignite is refreshed by Logoz
> [21:46:39.634] Logoz Pyroblast Alysrazor *5356*
> [21:46:39.913] Logoz Pyroblast Alysrazor * 47220*
> [21:46:40.144] Logoz Ignite Alysrazor 19144
> [21:46:40.451] Logoz Pyroblast Alysrazor *5450*
> [21:46:40.549] Alysrazor's Ignite is refreshed by Logoz
> [21:46:40.749] Logoz Pyroblast Alysrazor *47292*
> [21:46:41.288] Logoz Pyroblast Alysrazor *5451*
> [21:46:41.385] Alysrazor's Ignite is refreshed by Logoz
> [21:46:41.695] Logoz Pyroblast Alysrazor *47369*
> [21:46:42.090] Logoz Pyroblast Alysrazor *5450*
> [21:46:42.158] Logoz Ignite Alysrazor 32416
> [21:46:42.192] Alysrazor's Ignite is refreshed by Logoz
> [21:46:42.891] Logoz Pyroblast Alysrazor *46734*
> [21:46:42.891] Logoz Pyroblast Alysrazor *5450*
> [21:46:43.414] Alysrazor's Ignite is refreshed by Logoz
> [21:46:43.691] Logoz Pyroblast Alysrazor *5450*
> [21:46:43.691] Logoz Pyroblast Alysrazor *47104*
> [21:46:44.178] Logoz Ignite Alysrazor 51985
> [21:46:44.245] Alysrazor's Ignite is refreshed by Logoz
> Part of the log from http://worldoflogs.com/reports/rt-n57y9l5bwk25qlxi/log/?s=2286&e=2603
> Events we are interested in are highlighted in bold. The first highlighted refresh indicates when Ignite was extended beyond 4 seconds. The second highlighted event is the Pyroblast crit, which throws Ignite at the target (fourth event, refresh Ignit). The third event is the first tick of extended Ignit, which occurs only after Ignite is refreshed again by the crit above the 4-second duration. Other highlighted events repeat the same thing over and over again = Ignite refreshes even above 4 seconds duration.
> 
> 2) The so-called free tick is supposed to work here. If the mage critne about 0.6 - 0.2s before the Ignit tick, Ignite should be added to the pool, this next tick then ticks for the increased value, removes its part from the pool, but the subsequent refresh of Ignit returns this pool to its original value before the tick - a free tick is created.
> 
> [21:46:39.913] Logoz Pyroblast Alysrazor *47220*[21:46:40.144] Logoz Ignite Alysrazor 19144 - tick removes from pool[21:46:40.451] Logoz Pyroblast Alysrazor *5450*
> 
> 
> [21:46:40.549] Alysrazor's Ignite is refreshed by Logoz - the pool is restored to its original value before the Ignit tick
> Other tics are normally removed from the pool if the mage does not critne again in this time window before the next tick. This cannot be simulated in the current conditions, it is mainly used for Alysrazor in Firelands, where the player has increased haste and crit chance.
> 
> http://us.battle.net/wow/en/forum/topic/3195920164?page=3#43
> Here one of the players describes exactly how this free tick works. First of all, this free tick exists because of a "bug" where the new Ignite is not calculated by the server immediately, but with a delay of ~0.5 seconds. On Twinstar, Ignite is calculated at the same time as the crit that Ignite has passed, missing this delay that the offi servers had.
> 
> 
> https://www.youtube.com/watch?v=OOo-9p_-AXc https://www.youtube.com/watch?v=X6c3XXbbd3k
> In these two videos, you can see not only refreshing Ignit above the 4-second duration, but also stacking Ignit to a level that is currently not possible here, because the free tick does not work.
> 
> PS: Both of these mechanics also apply to [Deep Wounds](https://cata-twinhead.twinstar.cz/?issue=6062) (Fixed) and [Burning Embers (Fixed).](https://cata-twinhead.twinstar.cz/?issue=10528) So if it was possible to fix this with these two spells at the same time, it would be great (for DW and BE, maybe the 1st problem will work correctly, it's hard to test when these two DoT are ticking every second).

**Comment by firedragoon10** [comment]
that mean now fire mage will be  the bad spec in this extation so even retatil before and other server was buged the best  extation for fire mage  now become in this server trash classe  rely i cant understand  first  fire mage wth 4set and wth 30%  crit  stats  wt hraid buf and flask  36% 70% of bosses fight  overall damage fire  damage bewteen 28% and 30% and sometime you got fireball 40% overall damage crit or pyro or combuts when you see war doig  insane damage wth 17% CRIT ONE OF HIS SPELL doing 48% OVERALL damage crit  and now mage wth problem crit becasue wth nerf stack ignite wth crit damage rely i cant understand this server i think i will ask blizzard to make new catalysme server is better than play  here becasue fire mage in raid now wth 25% reduce damage tank wil ldo more kiki damage than him  gg wihtemane server

**Comment by firedragoon10** [comment]
> that mean now fire mage will be the bad spec in this extation so even retatil before and other server was buged the best extation for fire mage now become in this server trash classe rely i cant understand first fire mage wth 4set and wth 30% crit stats wt hraid buf and flask 36% 70% of bosses fight overall damage fire damage bewteen 28% and 30% and sometime you got fireball 40% overall damage crit or pyro or combuts when you see war doing insane damage wth 17% CRIT ONE OF HIS SPELL doing 48% OVERALL damage crit and now mage when he vae probelm  crit chance  he got  nerf stack ignite wth crit damage rely i cant understand this server i think i will ask blizzard to make new catalysme server is better than play here becasue fire mage in raid now wth 25% reduce damage tank wil ldo more kiki damage than him gg wihtemane server

**Comment by bonfax4** [comment]
Lol munching is present right now in classic wotlk, won't be any different if they decide to release cata. Curious but did anyone test deep wounds ? Is the dot not rolling and not cipping like this and instead stacking the same way ?

**Comment by Dechuka** [comment]
After applying latest fix, ignite stacking behavior is wrong,
check this :- 

**[21:46:37.466] Logoz Pyroblast Alysrazor 45238**
[21:46:37.990] Logoz Pyroblast Alysrazor 5355
[21:46:38.152] Alysrazor afflicted by Ignite from Logoz
[21:46:38.823] Logoz Pyroblast Alysrazor 5355
[21:46:38.823] Logoz Pyroblast Alysrazor 46823
[21:46:39.412] Alysrazor's Ignite is refreshed by Logoz
[21:46:39.634] Logoz Pyroblast Alysrazor 5356
[21:46:39.913] Logoz Pyroblast Alysrazor * 47220*
[21:46:40.144] Logoz Ignite Alysrazor 19144
[21:46:40.451] Logoz Pyroblast Alysrazor 5450
[21:46:40.549] Alysrazor's Ignite is refreshed by Logoz
[21:46:40.749] Logoz Pyroblast Alysrazor 47292
[21:46:41.288] Logoz Pyroblast Alysrazor 5451
[21:46:41.385] Alysrazor's Ignite is refreshed by Logoz
[21:46:41.695] Logoz Pyroblast Alysrazor 47369
[21:46:42.090] Logoz Pyroblast Alysrazor 5450
[21:46:42.158] Logoz Ignite Alysrazor 32416
[21:46:42.192] Alysrazor's Ignite is refreshed by Logoz
[21:46:42.891] Logoz Pyroblast Alysrazor 46734
[21:46:42.891] Logoz Pyroblast Alysrazor 5450
[21:46:43.414] Alysrazor's Ignite is refreshed by Logoz
[21:46:43.691] Logoz Pyroblast Alysrazor 5450
[21:46:43.691] Logoz Pyroblast Alysrazor 47104
**[21:46:44.178] Logoz Ignite Alysrazor 51985**
[21:46:44.245] Alysrazor's Ignite is refreshed by Logo

from the same log that was added above, in 6.779 seconds, the mage was able to build up ignite up to 51985

I did 17 pyroblast (all crits) and I was hitting them before first tick of ignite, (I'm not talking about free tick), I'm sure it took me more than 15 seconds to do it, and still I was not able to build ignite higher than 38745, to make it even worse, ** for testing purpose, my mastery provides 89% more DoT damage**, I'm sure that, the mage in the logs didn't have 89% more damage from mastery. in fact, I did some calculation in the video that I'll post it below, the mage mastery was around 37%

to make it even worse, I was hitting  around 51k per pyroblast, and the mage from logs was hitting around 46k

![log 1](https://github.com/gamefreedomgit/Maelstrom/assets/121187280/30a52b97-7158-40b7-adfa-ebfbfde08ca1)

![log 2](https://github.com/gamefreedomgit/Maelstrom/assets/121187280/fd07b74b-d5a0-4e09-932c-ee0930063e80)

![log 3](https://github.com/gamefreedomgit/Maelstrom/assets/121187280/fd6bdeb8-796c-46c8-b314-5b08cd490f64)

All that, I couldn't get close to his ignite damage

moreover, in this video https://www.youtube.com/watch?v=OOo-9p_-AXc that was also added in the comment above.
skip to 0:22, you will see that, the mage started to build up his ignite.

**went from 0 to 39.3k
        from 39.3k to 88.1k
        from 88.1k to 136.7k
        from 136.7k to 185.5k
        from 185.5k to 234.2k
        from 234.2 to 278.6k
        from 278.6k to 322.2k
        from 322.k to 362.9k**

so, 8 crits made around 362.9k, and the 17 pyroblast made 38745

Honestly, I don't know how ignite stacks at moment, but it should be like this (keep in mind that I'm not talking about free tick)

1- before Ignite first tick :-

: let's say ,first crit hits 1000, it provides total damage of ignite 400 (200 per tick),
when the second crit hits 1100 it provides 440 (220 per tick)
so it must refresh ignite and the new total damage of ignite will be 840 (420 per tick )*

to summarize it, 2 hits before the first tick, add their total ignite damage together (then divide them by 2.... for per tick)

2- After the first tick:
the first crit hits for 1000, it provides 400 (200 per tick)....... ignite's first tick will damage for 200
the second crit comes after the first tick, let's say the crit is 1100, which it provides 440 (220 per tick) so ignite will be refreshed and the total damage of ignite will be 640 (320 per tick

this report was made to explain it during PTR
https://github.com/gamefreedomgit/Maelstrom/issues/1380

**Comment by UltraNix** [comment]
`when the second crit hits 1100 it provides 440 (220 per tick)
so it must refresh ignite and the new total damage of ignite will be 840 (420 per tick )*`
You are wrong here, as the refreshed Ignite is now 6 sec duration and has 3 ticks. So it's 840/3 = 280 per tick.
```
Any refreshes at any time will refresh ignite to 6s in (max) duration (current duration is ignite's current duration mod 2 plus 4s): 3 ticks, one at 4s, one at 2s, one at 0s. Ignite's damage pool is divided evenly between these three ticks.
```

`the second crit comes after the first tick, let's say the crit is 1100, which it provides 440 (220 per tick) so ignite will be refreshed and the total damage of ignite will be 640 (320 per tick`
200 from previous tick + 440 from the new = 640 / 3 ticks = 213

**Comment by Dechuka** [comment]
@UltraNix 
1-  Actually it didn't happen as you said, you can see  the video above, the free tick called free for a reason :)
it's always divided by 2 not 3, otherwise ignite won't reach 360k in 6~ 8 crits
in addition, you may see also in the video, when the ignite was pushed to 6 seconds, the next crit hit on 5 sec, made the new ignite duration 4 seconds, and the next crit pushed it to 6 sec, it keeps going like that

2- regarding Ignite munching, and it happens when 2 crits hit almost at exact same time, not crit between 6 sec and 4sec, the mage was casting so fast, and almost had no travel time due to standing so close to the boss (referring to the mage in the video) 

these 2 points above easily can be confirmed in the video https://www.youtube.com/watch?v=OOo-9p_-AXc

munching  was a bug that Blizzard, couldn't/didn't want to look into it, because they wanted to change ignite mechanism, but I can't confirm if they changed its mechanism or not, so basically ignite munching is a bug not a feature.

**Comment by bonfax4** [comment]
> in addition, you may see also in the video, when the ignite was pushed to 6 seconds, the next crit hit on 5 sec, made the new ignite duration 4 seconds, and the next crit pushed it to 6 sec, it keeps going like that (no munching happened)

If you actually look at combustionhelper rather than the debuff on that video you see that the debuff is often refreshed to 4.0-4.5 seconds which would still show up as 4 secs on the debuff, it is not being refreshed to 4 seconds to be divided by 2 which doesn't make a lot of sense and you'd lose tick time in that case. Refreshes are still happening in this scenario and they are still being divided by 3 ticks, rather than 2. The debuff delay and the free ticks + alys buff would allow ignite to stagger enough and stack to the amounts you see in the video. The whole 6-4 second window munch is not a thing and it's pretty obvious when you see the mage dps alys close enough on the stacking phase.

Both or a mix of these scenarios is what would cause ignite to stack to those amounts since it would be constantly being reverted to, adding and delaying subtractions from the pool : 

Refresh delay window 
> [00:13.872] Spell Crit on Target
> [00:14.381] Ignite Tick
> [00:14.461] Ignite refreshed on Target
> 

and the "sandwhich"

> [00:30.064] Spell Crit 1 on Target
> [00:30.097] Spell Crit 2 on Target
> [00:30.512] Ignite refreshed on Target (from Spell Crit 1)
> [00:30.902] Ignite Tick
> [00:30.902] Ignite refreshed on Target (from Spell Crit 2)

So you'd need crits happening within up to 0.6 sec of a dot tick or whatever the refresh delay is often and in quick succession which is very likely with alysrazor haste and crit buff ( less than 1 sec pyros but locked to 1 sec gcd which would allow pyro + hot streak pyros to not munch each other because the gcd delay). Delay seems to be a bit random so i dunno how you would implement it here to begin with.

**Comment by UltraNix** [comment]
`regarding Ignite munching, and it happens when 2 crits hit almost at exact same time`
it's because on retail you had 500ms delay which is not present on our server and never will be.
Ignite Munching was a thing on cata retail, so the same is going on our server.

**Comment by Pornoodemon** [comment]
!['12'122'121'22'12'12'1](https://github.com/gamefreedomgit/Maelstrom/assets/133828891/cfcbb61a-9d66-4c74-ac8e-9a38855a1bc3)
If you put a small symbol (mirror сopy), then the calons begin to deal too much damage with the "Fireball" spell

**Comment by UltraNix** [comment]
@Dechuka So even if I crit again at 5.3 sec, it should still add damage to the pool and refresh duration of Ignite?

**Comment by bonfax4** [comment]
@UltraNix yes and you can see it in the alys hc [this](https://youtu.be/1tX64eLakrs) vid @ 5:20 and 10:05 stacking phases. The only difference is in retail if you critted @ 5.3 seconds the refresh would only happen around 4.8seconds and it would refresh back to 4.8 seconds ( 1.8 sec away from next tick ). Since we don't have that here a 5.3sec crit would refresh at and put the dot roughly at 5.3 if theres no delay.

Pay attention combustion helper and ignite timer and bar ( 2nd row ).Bar often refreshes to full with 4+ seconds on @5sec remaining crits.

**Comment by UltraNix** [comment]
But that way you bring back the bug reported here - you can then indefinetely stack Ignite (if you are lucky and crits do not happen in these 500ms delay windows).

**Comment by bonfax4** [comment]
@Ultranix no because it still has cata clip mechanics like any other dot. It shouldn't stack at all. Ignitee dmg "rolls" like deep wounds.

**Comment by bonfax4** [comment]
But unless you apply similar mechanics with delays and free ticks that don't subtract dmg from ignite pool you won't be seeing ignite stacking that high i think. But if we dont have double crit munch it might compensate for that anyway.

**Comment by UltraNix** [comment]
@bonfax4 Poke me on Discord.

**Comment by bonfax4** [comment]
@UltraNix Noticed that the double crit munch is using the 1st crit rather than the 2nd to calculate ignite which should not happen. This means if you cast a fireball crit > hot streak Pyroblast crit, Pyroblast ignite gets munched instead.

**Comment by UltraNix** [comment]
Fixed.
